components/pkg-export-kubernetes/src/manifest.rs

```diff
@@ -91,9 +95,26 @@ impl Manifest {
         };
         let name = format!("{}-{}", pkg_ident.name, version_suffix);
 
-        let image = match matches.value_of("IMAGE_NAME") {
+        let image_name = match matches.value_of("IMAGE_NAME") {
             Some(i) => i.to_string(),
-            None => pkg_ident.origin + "/" + &pkg_ident.name + ":latest",
+            None => {
+                let (image_name, tag) = match image {
+                    Some(i) => {
+                        (
+                            i.name().to_owned(),
+                            i.tags().get(0).unwrap_or(&"latest".to_owned()).clone(),
```

Comment:
Out of curiosity, why do you need the `&`? Isn't a `&str` already a borrow?

Also, why do you need the `clone`? Rust is so hard sometimes 😭

Reply:
`tags()` returns `&[String]` (reference to the slice of Strings). `get` on the slice returns something like `Option<&String>.

`unwrap_or` for `Option<T>` looks like `fn unwrap_or(self, def: T) -> T`, so in our case where `T` is `&String`, `unwrap_or` is `fn unwrap_or(self, def: &String) -> &String`. We need to pass a reference to String to `unwrap_or`.

`&` in `&"latest".to_owned()` borrows a reference not to `"latest"` but to the result of `"latest".to_owned()`. `to_owned()` clones and returns `String` so that's why prepend `&` there, because we need `&String` here.

`unwrap_or` returns `&String`, so this stuff probably causes lifetime issues, so we need to clone it to get `String`. So if there was not tag at all we are cloning the cloned string. Whoop dee doo.

Reply:
The double clone probably could be avoided with:
```
match i.tags().get(0) {
    Some(ref s) => s.clone(),
    None => "latest".to_owned(),
}
```

Reply:
Thanks for explaining @krnowak . But there is no actual double clone here. `&"latest.to_owned()` is only happening if value is None, so your match statement actually does exactly the same.

Reply:
Asked on IRC, this was presented as a better solution:

`i.tags().get(0).cloned().unwrap_or("latest".to_owned))`

Reply:
This solution sounds nice. Better than current one, because `"latest".to_owned()` is executed every time now - it is just a parameter to a function call.

Reply:
Forgot to mention that `cloned()` also allows us to remove the ugly `&` in `&"latest"` 👍